src/Razor/src/Microsoft.AspNetCore.Razor.Language/Legacy/CSharpCodeParser.cs

```diff
@@ -616,7 +636,7 @@ private void ParseStatement(in SyntaxListBuilder<RazorSyntaxNode> builder, Block
                         ParseCodeBlock(builder, block);
                         break;
                     case SyntaxKind.Keyword:
-                        if (!TryParseKeyword(builder, transition: null))
+                        if (!TryParseKeyword(builder, whitespace: null, transition: null))
```

Comment:
Why pass `null` here?

Reply:
Because we already accepted the whitespace if any in this case. This code path only happens if we are already in a statement block. This is also why transition is null. Because it can't exist.